### PR TITLE
Adding the scissor rect in the Batch and a way to enable the scissor during clear command

### PR DESCRIPTION
**Pull request by samcake**
Adding the scissor rect value in the batch so we can get rid of the glScissor calls

Adding a way to optionnally enable the scissor test just for the duration of the clear command in order to clear just a rect region of the framebuffer

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4ca333efe1f2f70f8a1794b50049e547383bd1b1 available at: https://gist.github.com/ded94dcc95fe7babcbbe<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5346.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5346.dmg

File: libraries/gpu/src/gpu/Batch.cpp [hunk 1]
```diff
@@ -102,9 +102,10 @@ void Batch::drawIndexedInstanced(uint32 nbInstances, Primitive primitiveType, ui
     _params.push_back(nbInstances);
 }
 
-void Batch::clearFramebuffer(Framebuffer::Masks targets, const Vec4& color, float depth, int stencil) {
+void Batch::clearFramebuffer(Framebuffer::Masks targets, const Vec4& color, float depth, int stencil, bool enableScissor) {
```
Comment (jherico) [inline libraries/gpu/src/gpu/Batch.cpp]: what if I want to set the scissor without clearing the framebuffer?

Comment (samcake) [inline libraries/gpu/src/gpu/Batch.cpp]: the drawcall are beeing applied scissor when scissr is enabled in the Pipeline/State

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8a269b697244cf8a989e660e3de78c2d1dd8dd92 available at: https://gist.github.com/ded94dcc95fe7babcbbe<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5346.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5346.dmg

**Pull request merged**